### PR TITLE
Issue #30579537: README and Docker updates related to deployment

### PR DESCRIPTION
**Pull request by dmurphy1**
## Updates

- Updates the README local Docker development instructions and the deployment instructions
- Updates the `Dockerfile` to copy the `js` directory into the Dockerized version of the app. See my inline comments.
- Updates the `package.json` file so `node start` runs `node` (i.e. production mode) and `node run dev` runs `nodemon` (i.e. development mode) since the Dockerized app runs the `start` command when the container starts (see `CMD [ "npm", "start" ]`) and we don't want to run `nodemom` on production. See my inline comments. I also updated the README accordingly.

## Testing

None. This is actually already deployed and the `Dockerfile` and `package.json` changes reflect updates I made when troubleshooting the deployment.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,6 +10,8 @@ RUN npm install
 
 # Bundle app source
 COPY app.js /usr/src/app/
+RUN mkdir -p /usr/src/app/js
+COPY js /usr/src/app/js
```
Comment (dmurphy1) [inline Dockerfile:14]: The `Dockerfile` is basically the instructions for building the "dockerized" version of the app (i.e. the image). These instructions are followed when we run `docker build` to build the image. Now that we have a `js` directory with dependencies, we need to copy it into the image. Before adding this `docker logs -f savas-slack-tools` showed an error starting with `Error: Cannot find module './js/single'` which helped me debug this.

File: package.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
   "main": "app.js",
   "scripts": {
     "test": "mocha -u exports ./test/*",
-    "start": "nodemon app.js"
+    "dev": "nodemon app.js",
+    "start": "node app.js"
```
Comment (dmurphy1) [inline package.json:9]: The last line of the `Dockerfile` is `CMD [ "npm", "start" ]` which is the command the container runs when it is started (based off an image). We don't think we want to run `nodemon` in production. I could have also updated the `Dockerfile` `CMD [ "npm", "start" ]`, but it seemed to me that it was better to have a `dev` script instead. I'm open to changes here though.

File: Dockerfile (hunk 1 above)
Comment (maddybvh) [reply Dockerfile:14]: Thanks for catching and fixing this!

File: package.json (hunk 2 above)
Comment (maddybvh) [reply package.json:9]: This makes sense to me! Good catch on avoiding nodemon on production 🤦

**Review by maddybvh — APPROVED** [review]
Thanks for making these updates @dmurphy1 ! Merging.

**Pull request merged**